File: kotlin_android_dev/app/src/main/kotlin/com/kotlin_android_dev/named_vm/app_vm/DataForAppVM.kt
package com.kotlin_android_dev.named_vm.app_vm

import io.github.antonpichka.lamm.base_data_for_named.BaseDataForNamed

class DataForAppVM(isLoading: Boolean): BaseDataForNamed<EnumDataForAppVM>(isLoading) {
    override fun getEnumDataForNamed(): EnumDataForAppVM {
        if(isLoading) {
            return EnumDataForAppVM.IS_LOADING
        }
        if(exceptionController.isWhereNotEqualsNullParameterException()) {
            return EnumDataForAppVM.EXCEPTION
        }
        return EnumDataForAppVM.SUCCESS
    }

    override fun toString(): String {
        return "DataForAppVM(isLoading: $isLoading, " +
                "exceptionController: $exceptionController)"
    }
}

File: dart_mobile_dev/lib/named_vm/app_vm/data_for_app_vm.dart
import 'package:library_architecture_mvvm_modify/library_architecture_mvvm_modify.dart';
import 'package:dart_mobile_dev/named_vm/app_vm/enum_data_for_app_vm.dart';

final class DataForAppVM extends BaseDataForNamed<EnumDataForAppVM> {
  DataForAppVM(super.isLoading);

  @override
  EnumDataForAppVM get getEnumDataForNamed {
    if(isLoading) {
      return EnumDataForAppVM.isLoading;
    }
    if(exceptionController.isWhereNotEqualsNullParameterException()) {
      return EnumDataForAppVM.exception;
    }
    return EnumDataForAppVM.success;
  }

  @override
  String toString() {
    return "DataForAppVM(isLoading: $isLoading, "
        "exceptionController: $exceptionController)";
  }
}